### PR TITLE
Bump to PHPUnit 8.5 and tests clean up

### PR DESCRIPTION
**Pull request by franmomu**
I was trying to add psalm and there were some issues in `Tests`, so this is a bit of clean up Tests before adding psalm. 

It is split in commits, the relevant ones:

- Psalm was complaining about some non-existing classes, I created some dummy ones.
- Update tests to be compatible with PHPUnit 8 and 9.
- ~[Drop support for doctrine/data-fixtures < 1.3, I could do this in a separate PR if it's necessary.](https://github.com/doctrine/DoctrineMongoDBBundle/pull/655)~

### CODE REVIEW

**Comment by franmomu** [comment]
Just commenting to have some activity.

**Comment by franmomu** [comment]
~Where did travis go? 🤔~ https://github.com/doctrine/DoctrineMongoDBBundle/pull/660

File: Tests/Mapping/Driver/AbstractDriverTest.php [hunk 1]
```diff
@@ -37,25 +38,23 @@ public function testFindMappingFileInSubnamespace()
...
     }
 
-    /**
-     * @expectedException Doctrine\Persistence\Mapping\MappingException
-     */
     public function testFindMappingFileNamespacedFoundFileNotFound()
     {
         $driver = $this->getDriver([$this->getFixtureDir() => 'MyNamespace\MyBundle\Document']);
 
+        $this->expectException(MappingException::class);
```
Comment (greg0ire) [inline Tests/Mapping/Driver/AbstractDriverTest.php]: This line should be moved right before the one that triggers the exception (so probably the last one, right?)

**Review by greg0ire — COMMENTED** [review]
Very nice, just one tiny nitpick.

**Review by malarzm — APPROVED** [review]

**Comment by malarzm** [comment]
Thanks @franmomu!

**Pull request merged**